sdk/servicebus/azure-servicebus/api_review.py

```diff
@@ -0,0 +1,248 @@
+        retry_backoff_factor: float = 30,
+        retry_backoff_maximum: int = 120  # cur_retry_backoff = retry_backoff_factor * (2^cur_retry_time)
+    ) -> None:
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str : str,
+        queue_name: str = None,
+        topic_name: str = None,
+        subscription_name : str = None,
+        session_id: str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        **kwargs
+    )-> ServiceBusReceiver:
+
+    def __enter__(self):
+    def __exit__(self):
+
+    def close(self) -> None:
+
+    def get_properties(self) -> Dict[str, any]:
+
+    def __iter__(self):
+    def __next__(self):
+    def next(self):
+
+    def peek(
+        self,
+        count : int = 1,
+        start_from : int = None
+    ) -> List[PeekMessage]:
+    def receive_deferred_messages(
+        self,
+        sequence_numbers : List[int],
+    ) -> List[DeferredMessage]:
+    def receive(
+        self,
+        max_batch_size : int = None,
+        timeout : float = None,
+    ) -> List[ReceivedMessage]:  # Pull mode receive
+
+    @property
+    def session(self) -> ServiceBusSession:
+
+    # Rule APIs, raise Error when called on Queue
+    def add_rule(self, rule_name : str, filter : str) -> None:
+    def remove_rule(self, rule_name : str) -> None:
+    def get_rules(self) -> List[str] -> Dict[str, Any]:
+
+
+class ServiceBusSession:
+    @property
+    def get_session_state(self) -> str:  # This can be made to property if property is more pythonic way
+    def set_session_state(self, state : str) -> None:  # This can be made to property if property is more pythonic way
+    def renew_lock(self) -> None:
+    @property
+    def session_id(self) -> str:
+    def expired(self) -> bool:
+
+
+class Transaction:
+    def __init__(self) -> None:
+    def __enter__(self):
+    def __exit__(self):
+    def begin(self) -> None:
+    def commit(self) -> None:
+    def abort(self) -> None:
+
+
+class Message:
+    def __init__(self, body : str, encoding : str = 'UTF-8', **kwargs) -> None:
+    def __str__(self):
+
+    # @properties
+    def body(self) -> Union[bytes, Generator[bytes]]:  # read-only
+    def partition_key(self, value : str):
+    def partition_key(self) -> str:
+    def via_partition_key(self, value: str):
+    def via_partition_key(self) -> str:
+    def time_to_live(self, value : Union[float, timedelta]):
+    def time_to_live(self) -> datetime.timedelta:
+    def annotations(self, value : dict[str, Any]):
+    def annotations(self) -> dict[str, Any]:
+    def user_properties(self, value : dict[str, Any]):
+    def user_properties(self) -> dict[str, Any]:
+    def enqueue_sequence_number(self, value : int):
+    def enqueue_sequence_number(self) -> int:
+
+    # Methods
+    def schedule(self, schedule_time_utc : datetime) -> None:
+
+
+class BatchMessage(Message):
+# inherited from Message
+    def add(self, message: Message) -> None:
```

Comment:
So if we're going with what we have in EH - this should be 'try_add' and it should return a bool.

Reply:
In EH we have `add`, because the behaviour is to raise if the add fails.